libs/util/httpRequest.js

```diff
@@ -76,17 +83,42 @@ function io(options, controller) {
 
             if (response.ok) {
                 return response.json().catch(function () {
-                    return null;
+                    throw new FetchrError(
+                        'BAD_JSON',
```

Comment:
I'm thinking it might be nice to move these error codes into a constants file and export them, that way we can also use them in the tests and not have to worry about typo errors in the future.

Reply:
I moved FetchrError to a separated module and set the reasons as static proprties in the class. I'm not sure if it wouldn't be better to have a separate class per kind of error instead of the reason field. But again, because of es5, doing inheritance is a pain in the ass...

Reply:
Yup makes sense to keep it ES5 compat for now. Changes look good.